src/openapi/transport/auth.js

```diff
@@ -6,150 +6,31 @@
-        this.tokenRefreshTimerFireTime = Date.now() + elapse;
-        this.tokenRefreshTimer = setTimeout(this.refreshOpenApiToken.bind(this), elapse);
-    }
-}
-
-function onApiTokenReceived(result) {
-    this.state = STATE_WAITING;
-    this.retries = 0;
-    if (!result.response || !result.response[this.tokenRefreshPropertyNameToken]) {
-        log.warn(LOG_AREA, 'Token refresh succeeded but no new token was present in response', result);
-        return;
-    }
-    const token = result.response[this.tokenRefreshPropertyNameToken];
-    const expiry = toAbsoluteTokenExpiry(result.response[this.tokenRefreshPropertyNameExpires]);
-    this.auth.set(token, expiry);
-    this.trigger(this.EVENT_TOKEN_RECEIVED, token, expiry);
-}
-
-function onApiTokenReceiveFail(result) {
-    log.warn(LOG_AREA, 'Token refresh failed', result);
-
-    if (result && (result.status === 401 || result.status === 403)) {
-        this.trigger(this.EVENT_TOKEN_REFRESH_FAILED);
-        this.state = STATE_FAILED;
-        return;
-    }
-
-    if (this.retries < this.maxRetryCount) {
-        this.state = STATE_WAITING;
-        this.retries++;
-        this.tokenRefreshTimerFireTime = Date.now() + this.retryDelayMs;
-        this.tokenRefreshTimer = setTimeout(this.refreshOpenApiToken.bind(this), this.retryDelayMs);
-    } else {
-        this.state = STATE_FAILED;
-        this.trigger(this.EVENT_TOKEN_REFRESH_FAILED);
-    }
-}
-
-function getToken(url) {
-    this.state = STATE_REFRESHING;
-    const headers = this.tokenRefreshHeaders || {};
-    headers['Content-Type'] = headers['Content-Type'] || 'JSON';
-
-    this.transport.fetch(this.tokenRefreshMethod, url, { headers, cache: false, credentials: this.tokenRefreshCredentials })
-        .then(onApiTokenReceived.bind(this), onApiTokenReceiveFail.bind(this));
-}
-
-function Auth(initialToken, initialExpiry, onChange) {
-    emitter.mixinTo(this);
-
-    function addBearer(newToken) {
-        if (newToken && !startsWith(newToken, TOKEN_BEARER, false)) {
-            newToken = TOKEN_BEARER + newToken;
-        }
-        return newToken;
-    }
-
-    let token = addBearer(initialToken);
-    let expiry = initialExpiry;
-    this.getToken = function() {
-        return token;
-    };
-    this.getExpiry = function() {
-        return expiry;
-    };
-    this.set = function(newToken, newExpiry) {
-
-        token = addBearer(newToken);
-        expiry = newExpiry;
-
-        onChange(token, expiry);
-        this.trigger('tokenReceived');
-    };
-}
-
 function makeTransportMethod(method) {
     return function(serviceGroup, urlTemplate, templateArgs, options) {
-        options = options || {};
-        options.headers = options.headers || {};
-        options.headers.Authorization = this.auth.getToken();
-
-        return this.transport[method](serviceGroup, urlTemplate, templateArgs, options)
-            .catch(onTransportError.bind(this, this.auth.getExpiry()));
+        const newOptions = {

```

Comment:
@michax just a guess - but I think this might be why we are getting the batch errors with different auth token - because a token is retried and ends up in a batch but this has mutated the headers

Reply:
Yeah, would match scenarios from live where we have both normal 401 failures followed by ones causes by subrequest containing auth token header.